src/client/components/SignupForm/index.js

```diff
@@ -3,7 +3,7 @@ import React from 'react';
 const SignupForm = (props) => (
   <form className="signup-form" method="POST" action="/api/signup"
     onSubmit={props.onSubmit}>
-    <input type="text" name="username"
+    <input type="text" name="username" pattern=".{4,10}"
```

Comment:
if it is required then we have to remove the Optional from the placeholder and add the required attribute to the field